Source/SwiftLintFramework/Rules/Style/SortedImportsRule.swift

```diff
@@ -173,6 +108,24 @@ struct SortedImportsRule: CorrectableRule, ConfigurationProviderRule, OptInRule
         }
     }
 
+    /// - returns: whether `lhs` should come before `rhs` based on a comparison of the contents of the import lines
+    private func should(_ lhs: Line, comeBefore rhs: Line) -> Bool {
+        switch configuration.grouping {
+        case .attributes:
+            let lhsAttributes = lhs.importAttributes()
+            let rhsAttributes = rhs.importAttributes()
+            if lhsAttributes != rhsAttributes {
+                if lhsAttributes.isEmpty != rhsAttributes.isEmpty {
```

Comment:
It's enough to check for `rhsAttributes.isEmpty` and, if so, return `true`. In other words, the whole `if` statement can be reduced to `return rhsAttributes.isEmpty || lhsAttributes < rhsAttributes`.

Reply:
Doing this fails the `  import AAA\n@testable import ↓BBB` test:

```swift
            let lhsAttributes = lhs.importAttributes()
            let rhsAttributes = rhs.importAttributes()
            if lhsAttributes != rhsAttributes {
                return rhsAttributes.isEmpty || lhsAttributes < rhsAttributes
            }
```

`rhsAttributes` is `@testable`, which is not empty, and `"" < "@testable"`. So it returns `true`. But we actually want it to return `false` to put attributes first.

Reply:
Got it. 👍